lib/urlwatch/reporters.py

```diff
@@ -668,18 +668,75 @@ def submit(self):
             summary.extend(summary_part)
             details.extend(details_part)
 
+        # Sum of the lengths of parts, with newlines taken into account
+        summary_len = sum(len(part) for part in summary) + len(summary) - 1
+
+        # Determine the space remaining after the summary. Reserve some extra
+        # space so we can at least report that all details were omitted if the
+        # message is too long.
+        remaining_len = max_length - summary_len - 40
+
+        if summary and show_footer:
+            footer = ('--- ',
+                      '%s %s, %s  ' % (urlwatch.pkgname, urlwatch.__version__, urlwatch.__copyright__),
+                      'Website: %s  ' % (urlwatch.__url__,),
+                      'watched %d URLs in %d seconds' % (len(self.job_states), self.duration.seconds))
+            footer_len = sum(len(part) for part in footer) + len(footer) - 1
+            remaining_len -= footer_len
+
+        # Calculate approximate available length per item, shared equally.
+        len_per_details = remaining_len // len(details)
```

Comment:
Would/should shorter summaries give back to the pool? e.g. if there's one change that only has a short summary, then `len_per_details` can in general be longer?

Reply:
At this point the length of the summaries should be accurately accounted for so they never take away more from the pool than they have to. The details do give back to the pool [here](https://github.com/thp/urlwatch/pull/572/files/38c15ac5004071f028e281c3ce657297fed863eb#diff-4a9994ac9199b1bd6143a8766ec1eb48R711), if they don't use the entirety of their allocated length.

Unless I misunderstood something.